opensquirrel/passes/merger/general_merger.py

```diff
@@ -147,50 +148,7 @@ def rearrange_barriers(ir: IR) -> None:
     ir.statements = flatten_list(statements_groups)
 
 
-def merge_single_qubit_gates(circuit: Circuit) -> None:
-    """Merge all consecutive 1-qubit gates in the circuit.
-
-    Gates obtained from merging other gates become anonymous gates.
-
-    Args:
-        circuit: Circuit to perform the merge on.
-    """
-    accumulators_per_qubit: dict[Qubit, BlochSphereRotation] = {
-        Qubit(qubit_index): I(qubit_index) for qubit_index in range(circuit.qubit_register_size)
-    }
-
-    statement_index = 0
-    while statement_index < len(circuit.ir.statements):
-        statement = circuit.ir.statements[statement_index]
-
-        # Accumulate consecutive Bloch sphere rotations
-        instruction: Instruction = cast(Instruction, statement)
-        if isinstance(instruction, BlochSphereRotation):
-            already_accumulated = accumulators_per_qubit[instruction.qubit]
-            composed = compose_bloch_sphere_rotations(instruction, already_accumulated)
-            accumulators_per_qubit[instruction.qubit] = composed
-            del circuit.ir.statements[statement_index]
-            continue
-
-        def insert_accumulated_bloch_sphere_rotations(qubits: list[Qubit]) -> None:
-            nonlocal statement_index
-            for qubit in qubits:
-                if not accumulators_per_qubit[qubit].is_identity():
-                    circuit.ir.statements.insert(statement_index, accumulators_per_qubit[qubit])
-                    accumulators_per_qubit[qubit] = I(qubit)
-                    statement_index += 1
-
-        # For barrier directives, insert all accumulated Bloch sphere rotations
-        # For other instructions, insert accumulated Bloch sphere rotations on qubits used by those instructions
-        # In any case, reset the dictionary entry for the inserted accumulated Bloch sphere rotations
-        if isinstance(instruction, Barrier):
-            insert_accumulated_bloch_sphere_rotations([Qubit(i) for i in range(circuit.qubit_register_size)])
-        else:
-            insert_accumulated_bloch_sphere_rotations(instruction.get_qubit_operands())
-        statement_index += 1
-
-    for accumulated_bloch_sphere_rotation in accumulators_per_qubit.values():
-        if not accumulated_bloch_sphere_rotation.is_identity():
-            if accumulated_bloch_sphere_rotation.is_anonymous:
-                accumulated_bloch_sphere_rotation = try_name_anonymous_bloch(accumulated_bloch_sphere_rotation)
-            circuit.ir.statements.append(accumulated_bloch_sphere_rotation)
+class Merger(ABC):
+    @abstractmethod
```

Comment:
I propose this method to be both abstract and static. You can use:

```
@staticmethod
@abstractmethod
```

as shown [here](https://stackoverflow.com/a/31590500/260313).